File: chronos/src/Chronos.ProfilerAgent/AgentSettings.cpp
#include "StdAfx.h"
#include "AgentSettings.h"

CAgentSettings::CAgentSettings(void)
{
}


CAgentSettings::~CAgentSettings(void)
{
	
}

__bool CAgentSettings::Initialize(CBaseStream* stream)
{
	//Seesion Token
	SessionToken = CStringMarshaler::Demarshal(stream);
	//CallPageSize
	stream->Read(&CallPageSize, TypeSize::_INT32);
	//ThreadStreamsCount
	stream->Read(&ThreadStreamsCount, TypeSize::_INT32);
	return true;
}

File: chronos/src/Chronos/Core/AgentSettings.cs
using System;

namespace Chronos.Core
{
	[Serializable]
	public class AgentSettings
	{
        public AgentSettings(Guid sessionToken, uint callPageSize, uint threadStreamsCount)
		{
			SessionToken = sessionToken;
			CallPageSize = callPageSize;
            ThreadStreamsCount = threadStreamsCount;
		}

		public Guid SessionToken { get; private set; }

		public uint CallPageSize { get; private set; }

        public uint ThreadStreamsCount { get; private set; }
	}
}
